src/main/java/org/jenkinsci/test/acceptance/plugins/ssh_slaves/SshSlaveLauncher.java

```diff
@@ -71,6 +77,20 @@ public SshSlaveLauncher keyCredentials(String username, String key) {
         cred.username.set(username);
         cred.enterDirectly(key);
         cred.add();
+        credentialUsername = username;
         return this;
     }
+
+    /**
+     * Once a credential has been created for a given slave, this method can be used 
+     * to check whether it has already been rendered in the dropdown.
+     */
+    public void waitForCredentialVisible() {
```

Comment:
I see, I prefer to make this part of the method that sets the value so the caller do not have to worry about this problem at all.